module/actor/calculations/stats-calculator.js

```diff
@@ -1,38 +1,103 @@
-    if (nature == "") return stat;
-    if (game.ptu.natureData[nature] == null) return statKey;
+function _calculateBaseStat(nature, statKey, edges, stat, abilities, level) {
+  if (nature == "") return stat;
+  if (game.ptu.natureData[nature] == null) return statKey;
 
-    if (game.ptu.natureData[nature][0] == statKey) stat += statKey == "HP" ? 1 : 2;
-    if (game.ptu.natureData[nature][1] == statKey) stat -= statKey == "HP" ? 1 : 2;
-    return Math.max(stat, 1);
+  if (game.ptu.natureData[nature][0] == statKey)
+    stat += statKey == "HP" ? 1 : 2;
+  if (game.ptu.natureData[nature][1] == statKey)
+    stat -= statKey == "HP" ? 1 : 2;
+
+  // Underdog's Strength
+  for (let edge of Object.entries(edges)) {
+    if (edge[1].name == "Underdog's Strength") stat += 1;
+  }
+
+  //Abilities
+  for (let ability of Object.entries(abilities)) {
+    //Abominable [Playtest]
+    if (ability[1].name == "Abominable [Playtest]" && statKey == "HP")
+      stat += 5;
+    //Huge Power or Pure Power
+    if (ability[1].name == "Huge Power" || ability[1].name == "Pure Power")
+      if (statKey == "Attack") stat += stat;
+    //Huge Power / Pure Power [Playtest]
+    if (ability[1].name == "Huge Power / Pure Power [Playtest]")
+      if (statKey == "Attack") stat += 5 + Math.floor(level / 10);
+    //Heavy Metal [Playtest]
+    if (ability[1].name == "Heavy Metal [Playtest]") {
+      if (statKey == "Defense") stat += 2;
+      if (statKey == "Speed") stat -= 2;
+    }
+    //Light Metal [Playtest]
+    if (ability[1].name == "Light Metal [Playtest]") {
+      if (statKey == "Defense") stat -= 2;
+      if (statKey == "Speed") stat += 2;
+    }
+    //Sorcery
+    if (ability[1].name == "Sorcery")
+      if (statKey == "Special Attack") stat += 5 + Math.floor(level / 10);
+  }
+
+  return Math.max(stat, 1);
 }
 
-export function CalculateStatTotal(levelUpPoints, stats) {
-    for (let [key, value] of Object.entries(stats)) {
-        let sub = value["value"] + value["mod"] + value["levelUp"];
-        levelUpPoints -= value["levelUp"];
-        if (value["stage"] > 0) {
-            value["total"] = Math.floor(sub * value["stage"] * 0.2 + sub);
-        } else {
-            if (key == "hp") {
-                value["total"] = sub;
-            } else {
-                value["total"] = Math.ceil(sub * value["stage"] * 0.1 + sub);
-            }
-        }
+export function CalculateStatTotal(levelUpPoints, stats, species, edges) {
+  for (let [key, value] of Object.entries(stats)) {
+    let sub = value["value"] + value["mod"] + value["levelUp"];
+    levelUpPoints -= value["levelUp"];
+    if (value["stage"] > 0) {
+      value["total"] = Math.floor(sub * value["stage"] * 0.2 + sub);
+    } else {
+      if (key == "hp") {
+        value["total"] = sub;
+      } else {
+        value["total"] = Math.ceil(sub * value["stage"] * 0.1 + sub);
+      }
+    }
+  }
+  //PokeEdges
+  //Check for Mixed Sweeper
+  for (let edge of edges)
```

Comment:
Looping over all edges twice here, can be made into one for loop & `continue` statement.